chain.go

```diff
@@ -126,6 +126,14 @@ func newChainWithConfig(name string, config Config) *chain {
 		c.context.Environment = newEnvironment(c)
```

Comment:
Let's add a test to `chain_test.go` that sets various reporters and checks how chain ctor fills context.TestingTB